tests/unit/test_main.py

```diff
@@ -125,12 +128,60 @@ def test_main_with_file_unchanged(
     save_pyproject,
 ):
     """Test file left unchanged."""
-    read_cli.return_value = TestArgs()
+    args = TestArgs()
+    read_cli.return_value = args
     read_config.return_value = pathlib.Path()
     parse_pyproject.return_value = {}
     reformat_pyproject.return_value = {}
 
     with OutputCapture() as output:
         main()
 
-    assert output.text == f"'{TestArgs.file}' left unchanged"
+    assert output.text == f"'{args.file}' left unchanged"
+
+
+@unittest.mock.patch("pyprojectsort.main.reformat_pyproject")
+@unittest.mock.patch("pyprojectsort.main._parse_pyproject_toml")
+@unittest.mock.patch("pyprojectsort.main._read_config_file")
+@unittest.mock.patch("pyprojectsort.main._read_cli")
+def test_check_option_reformat_needed(
+    read_cli,
+    read_config,
+    parse_pyproject,
+    reformat_pyproject
+):
+    """Test --check option when reformat occurs."""
+    args = TestArgs(check=True)
+    read_cli.return_value = args
+    read_config.return_value = pathlib.Path()
+    parse_pyproject.return_value = {}
+    reformat_pyproject.return_value = {"change": 1}
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        with OutputCapture() as output:
+            main()
+
+        assert output.text == f"'{args.file}' would be reformatted"
+        assert pytest_wrapped_e.value == 1
+
+
+@unittest.mock.patch("pyprojectsort.main.reformat_pyproject")
+@unittest.mock.patch("pyprojectsort.main._parse_pyproject_toml")
+@unittest.mock.patch("pyprojectsort.main._read_config_file")
+@unittest.mock.patch("pyprojectsort.main._read_cli")
+def test_check_option_reformat_not_needed(
+    read_cli,
+    read_config,
+    parse_pyproject,
+    reformat_pyproject
+):
+    """Test --check option when reformat is not needed."""
+    args = TestArgs(check=True)
+    read_cli.return_value = args
+    read_config.return_value = pathlib.Path()
+    parse_pyproject.return_value = {"unchanged": 1}
+    reformat_pyproject.return_value = {"unchanged": 1}
+    with OutputCapture() as output:
```

Comment:
Same as above, just a suggestion to add new line to match AAA pattern.

```suggestion
    reformat_pyproject.return_value = {"unchanged": 1}

    with OutputCapture() as output:
```